### ISSUE TITLE
Certain custom rule regular expressions aren't parsed properly

### ISSUE THREAD

**Issue by eyeballz** [realm/SwiftLint#717]
Hi,
i am trying to add custom warnings in my yaml to match `Void ->` and `-> ()`
The first rule to match `Void ->` and `Void->` works perfectly
```
custom_rules:
  void_as_parameter:
    included: "*.swift"
    name: "Void as Parameter"
    regex: "(Void ->)"
    message: "Use Void only as return type."
    severity: warning
```
but i cannot match `-> ()` and `->()`
i have tried `(->[ ]?\(\))` and `(\-\>[ ]?\(\))` but nothing works
Am I doing something wrong?

**Comment by jpsim** [comment]
Works for me. What version of SwiftLint do you have installed (`swiftlint version`)?

```shell
$ cat .swiftlint.yml 
custom_rules:
  void_as_parameter:
    included: "*.swift"
    name: "Void as Parameter"
    regex: "(Void ->)"
    message: "Use Void only as return type."
    severity: warning
$ cat main.swift 
let abc: Void -> () = {}
$ swiftlint
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'main.swift' (1/1)
/working_dir/main.swift:1:10: warning: Void as Parameter Violation: Use Void only as return type. (void_as_parameter)
Done linting! Found 1 violation, 0 serious in 1 file.
```

**Comment by eyeballz** [comment]
Sorry i was not clear enough, the posted rule works but the one to match ```-> ()``` does not.

```
$ cat .swiftlint.yml 
custom_rules:
  void_as_parameter:
    included: "*.swift"
    name: "Void as Parameter"
    regex: "(->[ ]?\(\))"
    message: "Use Void as return type instead of ()."
    severity: warning

$ cat main.swift 
let abc: () -> () = {}
```

**Comment by jpsim** [comment]
Oops, no you were clear in your issue I just misread it. I can reproduce this, I suspect it's the way we're creating `NSRegularExpression`s for custom rules. I'll mark this as a bug and hopefully will have time to take a closer look later.

**Comment by eyeballz** [comment]
should I submit a rule for this once the issue is resolved?

**Comment by jpsim** [comment]
Yes please!
